src/platform/Linux/bluez/ChipDeviceScanner.cpp

```diff
@@ -31,19 +31,44 @@
 namespace chip {
 namespace DeviceLayer {
 namespace Internal {
+
 namespace {
 
-struct GObjectUnref
+struct GDBusCancellableObjectManager
 {
-    template <typename T>
-    void operator()(T * value)
+    GCancellable * cancellable  = nullptr;
+    GDBusObjectManager * object = nullptr;
+
+    GDBusCancellableObjectManager() : cancellable(g_cancellable_new()) {}
+    ~GDBusCancellableObjectManager()
     {
-        g_object_unref(value);
+        if (cancellable != nullptr)
```

Comment:
under what circumstances will cancellable be null? Memory allocation error? Why do we have both object and cancellable? Some class doc comments would help regarding usage.

Why would we not just start cancellable with nullptr?

Reply:
> under what circumstances will cancellable be null?

I've used previous logic of cancellable usage and there was a check for cancellable object being null. However, after reading a little bit more about glib, it seems that `g_object_new` never fails, so `g_cancellable_new` also does not fail.

I've added simple comment, why this struct is required.